source/extensions/clusters/strict_dns/strict_dns_cluster.cc

```diff
@@ -28,16 +30,55 @@ StrictDnsClusterImpl::StrictDnsClusterImpl(const envoy::config::cluster::v3::Clu
                                            absl::Status& creation_status)
     : BaseDynamicClusterImpl(cluster, context, creation_status),
       load_assignment_(cluster.load_assignment()),
-      local_info_(context.serverFactoryContext().localInfo()), dns_resolver_(dns_resolver),
-      dns_refresh_rate_ms_(
-          std::chrono::milliseconds(PROTOBUF_GET_MS_OR_DEFAULT(cluster, dns_refresh_rate, 5000))),
-      dns_jitter_ms_(PROTOBUF_GET_MS_OR_DEFAULT(cluster, dns_jitter, 0)),
-      respect_dns_ttl_(cluster.respect_dns_ttl()) {
-  failure_backoff_strategy_ =
-      Config::Utility::prepareDnsRefreshStrategy<envoy::config::cluster::v3::Cluster>(
-          cluster, dns_refresh_rate_ms_.count(),
-          context.serverFactoryContext().api().randomGenerator());
+      local_info_(context.serverFactoryContext().localInfo()), dns_resolver_(dns_resolver) {
+  if (cluster.has_dns_config()) {
+    if (cluster.has_dns_refresh_rate() /* deprecated */) {
+      throw EnvoyException("Only one of dns_refresh_rate or dns_config can be specified.");
+    }
+    if (cluster.has_dns_jitter() /* deprecated */) {
+      throw EnvoyException("Only one of dns_jitter or dns_config can be specified.");
+    }
+    if (cluster.has_typed_dns_resolver_config() /* deprecated */) {
+      throw EnvoyException(
+          "Only one of typed_dns_resolution_config or dns_config can be specified.");
+    }
+    if (cluster.dns_lookup_family() != envoy::extensions::clusters::dns::v3::DnsConfig::AUTO &&
+        cluster.dns_config().dns_lookup_family() !=
+            envoy::extensions::clusters::dns::v3::DnsConfig::AUTO /* deprecated */) {
+      throw EnvoyException("Only one of dns_lookup_family or dns_config can be specified.");
+    }
+    if (cluster.has_dns_refresh_rate() /* deprecated */) {
+      throw EnvoyException("Only one of dns_refresh_rate or dns_config can be specified.");
+    }
+    if (cluster.has_dns_failure_refresh_rate() /* deprecated */) {
+      throw EnvoyException("Only one of dns_failure_refresh_rate or dns_config can be specified.");
+    }
```

Comment:
Probably want to put this in a separate function... That being said its not clear to me whether we want to do these checks and raise errors.